src/sentry/tasks/post_process.py

```diff
@@ -1368,15 +1364,32 @@ def detect_new_escalation(job: PostProcessJob):
     # Get escalation lock for this group. If we're unable to acquire this lock, another process is handling
     # this group at the same time. In that case, just exit early, no need to retry.
     lock = locks.get(f"detect_escalation:{group.id}", duration=10, name="detect_escalation")
+    extra = {
+        "org_id": group.organization.id,
+        "project_Id": job["event"].project.id,
+        "group_id": group.id,
+    }
     try:
         with lock.acquire():
-            project_escalation_rate = get_project_counts(group.project)
+            project_escalation_rate = get_latest_threshold(job["event"].project)
             group_hourly_event_rate = group.times_seen_with_pending / group_age_hours
-            if group_hourly_event_rate > project_escalation_rate:
+            # a rate of 0 means there was no threshold that could be calculated
+            if project_escalation_rate > 0 and group_hourly_event_rate > project_escalation_rate:
                 job["has_escalated"] = True
                 group.update(substatus=GroupSubStatus.ESCALATING)
                 add_group_to_inbox(group, GroupInboxReason.ESCALATING)
-    except UnableToAcquireLock:
+    except UnableToAcquireLock as error:
+        extra["error"] = error
+        logger.warning(
+            "tasks.post_process.detect_new_escalation.unable_to_acquire_lock", extra=extra
+        )
+        return
+    except Exception as error:
```

Comment:
if you do so, it means our pipeline default metric won't move https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/post_process.py#L704-L736 you also need separated monitoring for this.

I would prefer if you just re-raise

Reply:
@hubertsentry will make a PR to refactor, thanks for the tip!